pysteps/nowcasts/steps.py

```diff
@@ -587,12 +592,39 @@ def f(R, i):
     if measure_time:
         starttime_mainloop = time.time()
 
+    if isinstance(timesteps, int):
+        timesteps = range(timesteps + 1)
+        timestep_type = "int"
+    else:
+        original_timesteps = [0] + list(timesteps)
+        timesteps = nowcast_utils.binned_timesteps(original_timesteps)
+        timestep_type = "list"
+
     extrap_kwargs["return_displacement"] = True
+    R_f_prev = [R for i in range(n_ens_members)]
+    t_prev = [0.0 for j in range(n_ens_members)]
+    t_total = [0.0 for j in range(n_ens_members)]
 
     # iterate each time step
-    for t in range(n_timesteps):
-        print("Computing nowcast for time step %d... " % (t + 1), end="")
-        sys.stdout.flush()
+    for t in range(len(timesteps)):
```

Comment:
same as above
```suggestion
    for t, timestep in enumerate(timesteps):
```

Reply:
Done. Used subtimestep_idx instead of timestep because the latter is already used for another purpose.